File: e2e/scripts/browser_config.py
import streamlit as st
import time

if st.button("show main menu"):
    st.set_option("browser.hideMainMenu", False)
    st.experimental_rerun()

if st.button("hide main menu"):
    st.set_option("browser.hideMainMenu", True)
    st.experimental_rerun()

if st.button("show running icon"):
    st.set_option("browser.hideRunningIcon", False)
    st.experimental_rerun()

if st.button("hide running icon"):
    st.set_option("browser.hideRunningIcon", True)
    st.experimental_rerun()

time.sleep(0.5)

File: e2e/specs/browser_config.js
describe("browser.config", () => {
  before(() => {
    cy.visit("http://localhost:3000/");
  });

  it("main menu is hidden", () => {
    // click button to set hideMainMenu config option to true
    cy.get(".stButton > button")
      .contains("hide main menu")
      .click();
    cy.get(".MainMenu").should("not.exist");
  });

  it("main menu is shown", () => {
    // click button to set hideMainMenu config option to false
    cy.get(".stButton > button")
      .contains("show main menu")
      .click();
    cy.get(".MainMenu").should("exist");
  });
});
